### PR TITLE
22290 MimeConverter mimeEncode: aCollectionOrStream should return Stream and not String

### PR DESCRIPTION
**Pull request by astares**
https://pharo.fogbugz.com/f/cases/22290/MimeConverter-mimeEncode-aCollectionOrStream-should-return-Stream-and-not-String

- return a (read) stream now
- add a test
- cleanup assert:equals: in tests

### CODE REVIEW

File: src/Network-MIME/MimeConverter.class.st [hunk 1]
```diff
@@ -40,8 +40,8 @@ MimeConverter class >> mimeDecode: aStringOrStream to: outStream [
 
 { #category : #convenience }
 MimeConverter class >> mimeEncode: aCollectionOrStream [
-	^ String streamContents: [:out |
-		self mimeEncode: aCollectionOrStream to: out]
+	^ (String streamContents: [:out |
+		self mimeEncode: aCollectionOrStream to: out]) readStream
```
Comment (pavel-krivanek) [inline src/Network-MIME/MimeConverter.class.st:44]: it should be rather something like:
```
| writeStream |

writeStream := WriteStream on: String new.
self mimeEncode: aCollectionOrStream to: writeStream.
^ writeStream
```

It is definitely not efficient to create stream on string and then create a stream on the string content

**Comment by astares** [comment]
I will need to do a review on this - the failing tests seem to expect a string, not a stream

**Comment by Ducasse** [comment]
ok so let us know. What should we do with this PR. We close it now and when it is ready you do a new one?

**Comment by astares** [comment]
@Ducasse 

Sorry was away for a week. We need to decide if either 
 - Sabines topic of the bug is correct and a stream should be returned 
 - or a string should be returned. 

It looks like the failing tests expect a String while the PR (following the topic) return a stream

**Comment by svenvc** [comment]
Why would I want #mimeEncode: to return a stream and not a string ?
What am I supposed to do with a write stream as a result ?
This is silly.

**Comment by astares** [comment]
I have to talk to Sabine again - she opened the bug as she had trouble with german umlauts.
Thats why she suggested to solve it by returning a stream. 

We close this PR as I opened another one ONLY for the cleaning of the tests
https://pharo.fogbugz.com/f/cases/22804/Cleanup-MailMessageTest
and then we have to solve https://pharo.fogbugz.com/f/cases/22290
 in a different way.

**Comment by astares** [comment]
We can alternatively solve it by removing the #upToEnd call

This is now done in new PR  https://github.com/pharo-project/pharo/pull/2097

**Pull request closed without merging**